src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -296,4 +296,52 @@ bool Series::checkLevelInterval(unsigned int cycleDuration, unsigned int optimiz
     return true;
 }
 
+bool Series::validateCycle(unsigned int firstHourOfTheWeek, std::optional<double> initialLevel,
+        unsigned int cycleDuration) const
+{
+    if (initialLevel.has_value())
```

Comment:
Please add a comment like "Check that the initial level is inside the rule curves at all cycle timesteps"